heron/common/src/cpp/network/mempool.h

```diff
@@ -16,8 +16,10 @@
 #ifndef MEM_POOL_H
 #define MEM_POOL_H
 
+#include <google/protobuf/message.h>
 #include <vector>
 #include <unordered_map>
+#include <mutex>
 #include <typeindex>
 
 template<typename T>
```

Comment:
Not sure what the memory leak here is. Ideally would have had more comments.
I want to keep this change as logic less as possible so maybe we can find the leak and fix it in a separate pr.